### PR TITLE
Save R15 before global access

### LINKED ISSUE #254: Dynamic linking error on amd64
### PureGo Version

v0.7.1

### Operating System

- [ ] Windows
- [ ] macOS
- [X] Linux
- [ ] FreeBSD
- [ ] Android
- [ ] iOS

### Go Version (`go version`)

go version go1.22.4 linux/amd64

### What steps will reproduce the problem?

Build the following code with `-buildmode=plugin` option. (ref. https://pkg.go.dev/plugin)

```go
package main

import _ "github.com/ebitengine/purego"

func main() {}
```

`purego@v0.5.1` succeeds building, but `purego@v0.6.1` and `purego@v0.7.1` raise the error.

The complete reproduction code is [here](https://github.com/zoncoen-sample/purego-plugin-issue).
[sample error log](https://github.com/zoncoen-sample/purego-plugin-issue/actions/runs/9463761342/job/26069658602)

### What is the expected result?

`go build -buildmode=plugin -o plugin.so .` succeeds without error.

### What happens instead?

```
# github.com/ebitengine/purego
asm: sys_amd64.s:134: when dynamic linking, R15 is clobbered by a global variable access and is used here: 000[8](https://github.com/zoncoen-sample/purego-plugin-issue/actions/runs/9460766406/job/26060171519#step:4:9)6 (/home/runner/go/pkg/mod/github.com/ebitengine/purego@v0.7.1/sys_amd64.s:134)	MOVQ	R15, 32(SP)
asm: assembly failed
```

### Anything else you feel useful to add?

_No response_

### PR DESCRIPTION
**Pull request by TotallyGamerJet**
Closes #254

### CODE REVIEW

**Review by hajimehoshi — APPROVED** [review]
LGTM, thanks!

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -39,6 +39,12 @@ jobs:
         run: |
           go vet -v ./...
 
+      - name: go build plugin
+        if: runner.os == 'Linux'
```
Comment (hajimehoshi) [inline .github/workflows/test.yml:43]: plugin should work not only on Linux but also macOS and FreeBSD. Is there a reason why we specify Linux here?

Comment (TotallyGamerJet) [reply .github/workflows/test.yml:43]: I didn’t think it was necessary since they all behave the same but I can to be safe

Comment (hajimehoshi) [reply .github/workflows/test.yml:43]: Let's do this if it is easy.

**Review by hajimehoshi — APPROVED** [review]
Still LGTM

**Pull request merged**